src/pages/[platform]/build-a-backend/storage/storage-v5-to-v6-migration-guide/index.mdx

```diff
@@ -0,0 +1,1265 @@
+        },
+      });
+    }
+    ```
+
+  </Block>
+  <Block name="V5">
+    ```ts
+    import { Storage } from 'aws-amplify';
+
+    const handleGetUrl = async (key: string) => {
+      const url = await Storage.get(key, {
+        validateObjectExistence: true
+      });
+    }
+    ```
+
+  </Block>
+</BlockSwitcher>
+
+### Download File Content to Memory
+
+<BlockSwitcher>
+  <Block name="V6">
+    ```ts
+    import { downloadData } from 'aws-amplify/storage';
+
+    const handleDownload = async (key: string) => {
+      const { body, eTag } = await downloadData({
+        key,
+        options: {
+          accessLevel: 'guest'
+        }
+      }).result;
+    }
+    ```
+
+  </Block>
+  <Block name="V5">
+    ```ts
+    import { Storage } from 'aws-amplify';
+
+    const handleDownload = async (key: string) => {
+      const { Body, ETag } = await Storage.get(key, {
+        download: true,
+        validateObjectExistence: true,
+        progressCallback: ({ loaded, total }) => {
+          // Progress implementation
+        }
+      });
+    }
+    ```
+
+  </Block>
+</BlockSwitcher>
+
+### Monitor Download Progress
+
+The `progressCallback` option has been replaced by `onProgress` in v6. The function input property names `loaded` and `total` now map to `transferredBytes` and `totalBytes`.
+
+<BlockSwitcher>
+  <Block name="V6">
+  ```
+  import { downloadData } from 'aws-amplify/storage';
+
+  const handleDownload = async (key: string) => {
+    const { body, eTag } = await downloadData({
+      key,
+      options: {
+        accessLevel: 'guest',
+        onProgress: ({ transferredBytes, totalBytes }) => {
+          // Progress implementation
+        }
+      }
+    }).result;
+  }
+  ```
+
+  </Block>
+  <Block name="V5">
+    ```
+    import { Storage } from 'aws-amplify';
+
+    const handleDownload = async (key: string) => {
+      const { Body, ETag } = await Storage.get(key, {
+        download: true,
+        validateObjectExistence: true
+      });
+    }
+    ```
+
+  </Block>
+</BlockSwitcher>
+
+## Storage.copy
+
+In addition to [the changes mentioned above](/[platform]/build-a-backend/storage/storage-v5-to-v6-migration-guide/#deprecated-options), the `copy` API in v6 no longer supports content options (`contentDisposition`, `contentLanguage`, and `contentType`) as they override options selected on the original file `upload` and have caused complexity in implementation.
+
+<Accordion title='Input' headingLevel='3'>
+  <Columns columns={2}>
+    <div>
+      **V5**
+
+      ```
+      // Assumes default provider 'AWSS3': custom provider not supported in v6
+      src: {
+        key: string;
+        level?: string;
+        identityId?: string;
+      }
+      dest: {
+        level?: string;
+        key: string;
+      }
+      config?: {
+        provider?: 'AWSS3';
+        customPrefix?: {
+          public?: string;
+          protected?: string;
+          private?: string;
+        };
+        track?: boolean;
```

Comment:
Are all these still supported in v6? I noticed these options had comments in APIs above - should they be here also?

Reply:
Good catch, forgot to move those comments down and apply across the board.